stdlib/3/builtins.pyi

```diff
@@ -278,15 +278,15 @@ class str(Sequence[str]):
     def strip(self, chars: Optional[str] = None) -> str: ...
     def swapcase(self) -> str: ...
     def title(self) -> str: ...
-    def translate(self, table: Dict[int, Any]) -> str: ...
+    def translate(self, table: Union[Mapping[int, Union[int, str, None]], Sequence[Union[int, str, None]]]) -> str: ...
     def upper(self) -> str: ...
     def zfill(self, width: int) -> str: ...
     @staticmethod
     @overload
-    def maketrans(x: Union[Dict[int, Any], Dict[str, Any]]) -> Dict[int, Any]: ...
+    def maketrans(x: Mapping[Union[int, str], _T]) -> Dict[int, _T]: ...
```

Comment:
Unfortunately, this function does in fact require a Dict as the argument: the C code does `PyDict_CheckExact()` on the argument and errors out if it's something else.

Reply:
That's unfortunate indeed. So, if this needs to be a `Dict` (as is also specified in the docs), can you explain to me why invariance would prevent us from passing `Dict[int, _T]`? I'm not seeing the problem tbh.

Reply:
If the function expects a `Dict[Union[str, int], _T]`, then it may do something like `x['some_string'] = ...`. That would make code like this unsafe:
```
x: Dict[int, int] = {0: 0}
str.maketrans(x)
for k in x.keys():
    assert isinstance(k, int)  # suddenly there is a str
```
We know that `maketrans` doesn't actually modify the mapping, but mypy doesn't.

I think a working solution is to declare the argument type `Union[Dict[int, _T], Dict[str, _T]]`.

Reply:
Right, forgot for a moment that dictionaries are mutable.

So, to include the case when the dict has both integers and strings as keys, which is supported, I could add that to the Union as well, right? Turning the type into this monstrosity: `Union[Dict[int, _T], Dict[str, _T], Dict[Union[str, int], _T]`.

Reply:
Yes.